src/post/content.ts

```diff
@@ -207,6 +211,77 @@ export class ContentPreparer {
+            const escapedHandle = mention.name.replace(
+                /[.*+?^=!:${}()|\[\]\/\\]/g,
+                '\\$&',
+            );
+
+            // Handling existing hyperlinks in content
+            const existingLinkRegex = new RegExp(
+                `<a[^>]*href=["'](?:${escapedAccountApId}|${escapedAccountUrl})["'][^>]*>.*?</a>`,
+                'g',
+            );
+
+            // Replace existing hyperlinks with newly constructed hyperlinks with data-profile and rel attributes
+            preparedContent = preparedContent.replace(
+                existingLinkRegex,
+                (match) => {
+                    // Extract the href value from the existing link
+                    const hrefMatch = match.match(/href=["']([^"']*)["']/);
+                    const originalHref = hrefMatch
+                        ? hrefMatch[1]
+                        : mention.account.url;
+
+                    // Extract and clean up other attributes from the existing link
+                    const existingAttrs = match
+                        .replace(/<a\s+/, '') // Remove the opening <a tag
+                        .replace(/>.*$/, '') // Remove everything after the > (including the content and closing tag)
+                        .replace(/href=["'][^"']*["']/, '') // Remove the href attribute
+                        .replace(/rel=["'][^"']*["']/, '') // Remove the rel attribute if exists
+                        .replace(/data-profile=["'][^"']*["']/, '') // Remove the data-profile attribute if exists
+                        .replace(/\s+/g, ' ') // Normalize whitespace to single spaces
+                        .trim(); // Remove leading/trailing whitespace
+
+                    // Get the original content between the <a> tags
+                    const linkContent = match.replace(/<a[^>]*>|<\/a>/g, '');
+
+                    // Constructing new link
+                    const newHyperLink = `<a href="${originalHref}" data-profile="@${mention.account.username}@${mention.account.apId.hostname}" rel="nofollow noopener noreferrer"${existingAttrs ? ` ${existingAttrs}` : ''}>${linkContent}</a>`;
+                    return newHyperLink;
+                },
+            );
+
+            // Checking for mentions that aren't already wrapped in a hyperlink
+            const hasExistingLink =
+                preparedContent.includes(`<a href="${mention.account.url}"`) ||
+                preparedContent.includes(`<a href="${mention.account.apId}"`);
+
+            if (!hasExistingLink) {
+                const unwrappedMentionRegex = new RegExp(escapedHandle, 'g');
+                preparedContent = preparedContent.replace(
+                    unwrappedMentionRegex,
+                    `<a href="${mention.account.url}" data-profile="@${mention.account.username}@${mention.account.apId.hostname}" rel="nofollow noopener noreferrer">${mention.name}</a>`,
+                );
+            }
+        }
```

Comment:
This part feels a bit risky - if our hyperlink matching logic failed somehow, then we may be constructing invalid HTML (e.g. nested hyperlinks)?

Reply:
I wonder if we should drop the requirement to wrap plaintext mentions into hyperlinks for now: Mastodon sends us mentions in hyperlinks, and we don't know which AP implementations sends plaintext mentions.

This feels easier / less risky to ship?

We can always back to this in context and then decide whether we want to support plaintext mentions or get the AP implementation to fix their context

Reply:
> then we may be constructing invalid HTML (e.g. nested hyperlinks)

You are right about this. I faced this while writing tests. Test case: if the added link is not an apId or url, this part of code will add a nested hyperlink.

Generally if we aren't sure if there is any AP implementations that sends plaintext mentions, I am happy to drop this requirement


> This feels easier / less risky to ship?

💯

Reply:
Removed the support for plaintext mentions for now